### PR TITLE
Update unit tests to work with Pester 5.2.1

### PR DESCRIPTION
**Pull request by N-lson**

### CODE REVIEW

File: OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1 [hunk 1]
```diff
@@ -67,7 +65,7 @@ Describe "Invoke-OctopusServerCommand" {
     Context "It should not leak passwords" {
 
         BeforeAll {
-            . $script:modulePath
+            # . $script:modulePath
```
Comment (N-lson) [inline OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1:68]: These comments were removed in a later commit which also included Exe changes so I've left them as is for now.

File: OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1 [hunk 2]
```diff
@@ -1,16 +1,14 @@
-$moduleName = Split-Path ($PSCommandPath -replace '\.Tests\.ps1$', '') -Leaf
-$modulePath = Split-Path $PSCommandPath -Parent
-$script:modulePath = Resolve-Path "$PSCommandPath/../../$moduleName.ps1"
+BeforeAll {
```
Comment (N-lson) [inline OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1:1]: https://pester-docs.netlify.app/docs/usage/test-file-structure
```
Put all your code into It, BeforeAll, BeforeEach, AfterAll or AfterEach. Put no code directly into Describe, Context or on the top of your file, without wrapping it in one of these blocks, unless you have a good reason to do so.

All misplaced code will run during Discovery, and its results won't be available during Run which will lead to confusing results.

See Discovery and Run.
```

File: OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1 (hunk 1 above)
Comment (matt-richardson) [reply OctopusDSC/Tests/OctopusDSCHelpers.Tests.ps1:68]: Given we're looking for self-contained PRs (which will become self-contained commits to `master`), we should not intertwine changes between PRs. It's better practice to fix this here.

**Review by matt-richardson — CHANGES_REQUESTED** [review]
Thanks for sorting this out! 

I would've expected to see some changes to the launcher files (eg [build-aws.ps1](https://github.com/OctopusDeploy/OctopusDSC/blob/master/build-aws.ps1#L46)) to update the expected version of Pester.

**Review by matt-richardson — APPROVED** [review]
Nice one - thanks for that!

**Review by matt-richardson — CHANGES_REQUESTED** [review]
Ooh - something's gone wrong - this is only reporting 880 tests, but there should be 1625 tests.

File: Tests/OctopusDSC.Tests.ps1 [hunk 3]
```diff
@@ -1,239 +1,268 @@
-$script:scriptRoot = $PSScriptRoot
-
-Describe "PSScriptAnalyzer" {
-    BeforeAll {
+Describe "OctopusDSC" {
+    BeforeDiscovery {
```
Comment (matt-richardson) [inline Tests/OctopusDSC.Tests.ps1:2]: Ooh - never head of this one

File: Tests/OctopusDSC.Tests.ps1 [hunk 4]
```diff
@@ -1,239 +1,268 @@
...
+                $parseErrorCount = $parseErrors.length
+                $parseErrorCount | should -be 0
             }
         }
     }
 
-    BeforeAll {
-        . (Join-Path -Path $script:scriptRoot -ChildPath "powershell-helpers.ps1")
-    }
+    Describe "Mandatory and non-mandatory parameters" {
```
Comment (matt-richardson) [inline Tests/OctopusDSC.Tests.ps1:129]: Consider splitting this into two cases - they are separate enough now.

**Review by matt-richardson — APPROVED** [review]
Nice work - looks great.
Impressed you've managed to dive in and navigate your way through this.
Added one minor suggestion for consideration - will leave it up to you whether to do it or not.

File: Tests/OctopusDSC.Tests.ps1 (hunk 3 above)
Comment (N-lson) [reply Tests/OctopusDSC.Tests.ps1:2]: Looks new in v5 https://pester-docs.netlify.app/docs/usage/discovery-and-run
TL;DR the test cases are generated in the discovery phase but you can't declare variables and such in this phase as they're discarded
